integrationtests/tests/fixtures/utils.rs

```diff
@@ -0,0 +1,56 @@
+use async_trait::async_trait;
+use ln_gateway::ln::{LightningError, LnRpc};
+use std::collections::HashMap;
+use std::sync::Arc;
+use tokio::sync::Mutex;
+
+/// A proxy for the underlying LnRpc which can be used to add behavoir to it using the "Decorator pattern"
+pub struct LnRpcAdapter {
+    /// The actual LnRpc that we add behavior to.
+    client: Box<dyn LnRpc>,
+    /// A pair of <Invoice> and <Count> where client.pay() will fail <Count> times for each <Invoice>
+    fail_invoices: Arc<Mutex<HashMap<String, u8>>>,
```

Comment:
Can you use a `HashMap<String, AtomicU8>`?  Makes for less verbose usage if possible.